src/o365/graph/commands/teams/teams-channel-set.ts

```diff
@@ -0,0 +1,149 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import { Channel } from './Channel';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelName: string;
+  newChannelName: string;
+  description: string
+}
+class GraphTeamsChannelSetCommand extends GraphItemsListCommand<Channel>{
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_SET}`;
+  }
+  public get description(): string {
+    return 'Updates properties of a specified channel in the given Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.role = args.options.teamId;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let channelId: string = '';
+    const endpoint: string = `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels`;
+    this.getAllItems(endpoint, cmd, true)
+      .then((): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Channels in current Microsoft Teams team')
+          cmd.log(this.items);
+          cmd.log('');
+        }
+        const channelItem: Channel | undefined = this.items.find(c => c.displayName && c.id ? (c.displayName.toLowerCase() === args.options.channelName.toLowerCase()) : false);
+        if (channelItem) {
+          if (this.debug) {
+            cmd.log('The specified channel to be updated')
+            cmd.log(channelItem);
+            cmd.log('');
+          }
+          channelId = channelItem.id;
+        }
+        else {
+          throw new Error(`The specified channel does not exist in the Microsoft Teams team`);
+        }
+        const endpoint: string = `${auth.service.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels/${channelId}`;
+        const requestOptions: any = {
+          url: endpoint,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            'accept': 'application/json;odata.metadata=none'
+          }),
+          json: true,
+          body: { "description": args.options.description, "displayName": args.options.newChannelName }
```

Comment:
We should only specify properties that have been set, to prevent resetting values that haven't been set